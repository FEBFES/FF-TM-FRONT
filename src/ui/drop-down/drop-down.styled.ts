import styled from 'styled-components';

export const SDropDownContainer = styled.div`
  position: absolute;
  right: 0;
  top: 0;
  background: var(--bg-0);
  border: 1px solid var(--bg-border);
  padding: 8px 5px;
  border-radius: 8px;

  & > div {
    margin-bottom: 10px;
  }

  & > div:last-of-type {
    margin-bottom: 0;
  }
`;
