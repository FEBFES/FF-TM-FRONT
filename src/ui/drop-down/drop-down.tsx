import React from 'react';
import { useClickOutside } from '../../hooks/useClickOutside';
import { SDropDownContainer } from './drop-down.styled';

interface DropDownProps {
  children: React.ReactNode;
  setShow: (bool: boolean) => void;
  show: boolean;
}

export const DropDown: React.FC<DropDownProps> = ({
  children,
  setShow,
  show,
}) => {
  const { ref } = useClickOutside(setShow);

  if (!children || !show) {
    return null;
  }

  return (
    <SDropDownContainer
      id={'DropDown'}
      onClick={(e) => {
        e.stopPropagation();
      }}
      ref={ref}
    >
      {children}
    </SDropDownContainer>
  );
};
