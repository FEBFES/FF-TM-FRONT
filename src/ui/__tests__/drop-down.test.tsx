import React from 'react';
import { DropDown } from '../drop-down/drop-down';
import renderer from 'react-test-renderer';

it('SNAPSHOTS: UI - DropDown component - render', () => {
  const elem = renderer
    .create(
      <DropDown show={true} setShow={() => {}}>
        <div>Hellow</div>
        <div>Hi</div>
      </DropDown>
    )
    .toJSON();
  expect(elem).toMatchSnapshot();
});
