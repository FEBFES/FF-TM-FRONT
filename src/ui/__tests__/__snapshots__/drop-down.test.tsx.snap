// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOTS: UI - DropDown component - render 1`] = `
<div
  className="sc-bcPKhP bivSfu"
  id="DropDown"
  onClick={[Function]}
>
  <div>
    Hellow
  </div>
  <div>
    Hi
  </div>
</div>
`;
